### PR TITLE
Fix link not exist error.

### PR DESCRIPTION
**Pull request by chenrujun**
Fix link not exist error: https://dev.azure.com/azure-sdk/internal/_build/results?buildId=992339&view=logs&j=b70e5e73-bbb6-5567-0939-8415943fadb9&t=aea0e0e1-1e00-5c1b-bba8-b2b238976da3&l=19

### CODE REVIEW

**Comment by chenrujun** [comment]
Hi, @yiliuTo ， please help to review this PR.

File: sdk/spring/azure-spring-cloud-autoconfigure/README.md [hunk 1]
```diff
@@ -91,7 +91,7 @@ Please follow [instructions here][contributing_md] to build from source or contr
 [event_hubs]: https://github.com/Azure/azure-sdk-for-java/tree/main/sdk/spring/azure-spring-cloud-starter-eventhubs
 [event_hubs_kafka]: https://github.com/Azure/azure-sdk-for-java/tree/main/sdk/spring/azure-spring-cloud-starter-eventhubs-kafka
 [service_bus]: https://github.com/Azure/azure-sdk-for-java/tree/main/sdk/spring/azure-spring-cloud-starter-servicebus
-[storage]: https://github.com/Azure/azure-sdk-for-java/tree/main/sdk/spring/azure-spring-starter-storage
+[storage]: https://github.com/Azure/azure-sdk-for-java/tree/main/sdk/spring/azure-spring-boot-starter-storage
```
Comment (yiliuTo) [inline sdk/spring/azure-spring-cloud-autoconfigure/README.md:94]: Not sure if Azure Spring Boot Storage starter is suitable to be here, I think we could move this entry to azure-spring-boot's README?

Comment (chenrujun) [reply sdk/spring/azure-spring-cloud-autoconfigure/README.md]: Done.

**Review by yiliuTo — APPROVED** [review]
LGTM

**Pull request merged**